src/fstab-generator/fstab-generator.c

```diff
@@ -679,148 +833,15 @@ static int parse_fstab(bool initrd) {
-                        k = add_swap(fstab, what, me, flags);
-                else {
-                        bool rw_only, automount, is_sysroot, is_sysroot_usr;
-
-                        rw_only = fstab_test_option(me->mnt_opts, "x-systemd.rw-only\0");
-                        automount = fstab_test_option(me->mnt_opts,
-                                                      "comment=systemd.automount\0"
-                                                      "x-systemd.automount\0");
-
-                        flags |= rw_only * MOUNT_RW_ONLY |
-                                 automount * MOUNT_AUTOMOUNT;
-
-                        is_sysroot = in_initrd() && path_equal(where, "/sysroot");
-                        /* See comment from add_sysroot_usr_mount about the need for extra indirection
-                         * in case /usr needs to be mounted in order for the root fs to be synthesized
-                         * based on configuration included in /usr/, e.g. systemd-repart. */
-                        is_sysroot_usr = in_initrd() && path_equal(where, "/sysroot/usr");
-
-                        const char *target_unit =
-                                initrd ?               SPECIAL_INITRD_FS_TARGET :
-                                is_sysroot ?           SPECIAL_INITRD_ROOT_FS_TARGET :
-                                is_sysroot_usr ?       SPECIAL_INITRD_USR_FS_TARGET :
-                                mount_is_network(me) ? SPECIAL_REMOTE_FS_TARGET :
-                                                       SPECIAL_LOCAL_FS_TARGET;
-
-                        if (is_sysroot && is_device_path(what)) {
-                                r = generator_write_initrd_root_device_deps(arg_dest, what);
-                                if (r < 0)
-                                        return r;
-                        }
-
-                        k = add_mount(fstab,
-                                      arg_dest,
-                                      what,
-                                      is_sysroot_usr ? "/sysusr/usr" : canonical_where ?: where,
-                                      !is_sysroot_usr && canonical_where ? where : NULL,
-                                      me->mnt_type,
-                                      me->mnt_opts,
-                                      me->mnt_passno,
-                                      flags,
-                                      target_unit);
-
-                        if (is_sysroot_usr && k >= 0) {
-                                log_debug("Synthesizing fstab entry what=/sysusr/usr where=/sysroot/usr opts=bind");
-
-                                r = add_sysusr_sysroot_usr_bind_mount(fstab);
-                                if (r != 0)
-                                        k = r;
-                        }
-                }
-
-                if (arg_sysroot_check && k > 0)
+        for (struct mntent *me; (me = getmntent(f));) {
```

Comment:
using a for() loop for this looks weird. You are writing a while() loop here, but then mangle it into a for(). This does not help readability. There's no shame in writing this as:

```c
struct mntent *me;
while ((me = getmntent(f)) {
```

that's a ton more obvious and easier to read.